app/javascript/components/pages/Analytics.js

```diff
@@ -1,10 +1,99 @@
-import React, {Component} from 'react'
-import ReactDOM  from 'react-dom'
+import React, { Component } from 'react';
+import { getChannelData, getVideoData, getVideoStatData } from './api/yt_api'
+import Videos from './api/Videos'
+import Snippet from './api/Snippet'
+import Statistics from './api/Statistics'
 
-class Analytics extends Component{
-    render(){
-        return(
-            <h1>Analytics</h1>)
+
+class Analytics extends Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      data: null,
+      channelId: "UC-lHJZR3Gqxm24_Vd_AJ5Yw",
+      videoData: null,
+      videoId: null,
+      renderStr: "",
+      isLoaded: null,
     }
+  }
+  getVideoStat = () => {
+    let { videoData } = this.state
+    let videoIds = []
+
+    videoData.map((video, index) => {
```

Comment:
map will return an array, so you are doing extra work here.  22-26 canjust be:

`let videoIds = videoData.map((video)=> video.contentdetails.videoId )`